js/extensions/bindings/alphaNumericExtender.js

```diff
@@ -0,0 +1,25 @@
+define(['knockout'], function (ko) {
+    ko.extenders.alphaNumeric = function(target) {
+        //create a writable computed observable to intercept writes to our observable
+        var result = ko.pureComputed({
+            read: target,  //always return the original observables value
+            write: function(newValue) {
+                const isAlphaNumeric = RegExp('^[a-zA-Z0-9.]+$');
+                let current = target();
+    
+                //only write if it changed
+                if (isAlphaNumeric.test(newValue)) {
+					target(newValue);
+                } else {
+					target.notifySubscribers(current);
```

Comment:
@anthonysena , why do we need to notify subscribers if the value wasn't changed?

Reply:
The `newValue` may contain non-alphaNumeric values and so it is tested on line 11. If that fails, we want to keep the `current` value. This is done by notifying the original subscriber (`target`) that the value has changed (back to `current`) which effectively resets the observable and removes the non-alphaNumeric character.

If I take out the `notifySubscribers` call, the observable continues to retain the non-alphaNumeric values which is not the desired functionality.